sentry-core/src/main/java/io/sentry/core/EnvelopeSender.java

```diff
@@ -0,0 +1,107 @@
+    this.hub = Objects.requireNonNull(hub, "Hub is required.");
+    this.envelopeReader = Objects.requireNonNull(envelopeReader, "Envelope reader is required.");
+    this.serializer = Objects.requireNonNull(serializer, "Serializer is required.");
+    this.logger = Objects.requireNonNull(logger, "Logger is required.");
+  }
+
+  @Override
+  public void processEnvelopeFile(@NotNull String path) {
+    InputStream stream = null;
+    File file = null;
+    try {
+      file = new File(path);
+      stream = new FileInputStream(file);
+      SentryEnvelope envelope = envelopeReader.read(stream);
+      if (envelope == null) {
+        logger.log(SentryLevel.ERROR, "Stream from path %s resulted in a null envelope.", path);
+      } else {
+        processEnvelope(envelope);
+      }
+    } catch (Exception e) {
+      logger.log(SentryLevel.ERROR, "Error processing envelope.", e);
+    } finally {
+      try {
+        if (stream != null) stream.close();
+      } catch (IOException ex) {
+        logger.log(SentryLevel.ERROR, "Error closing envelope.", ex);
+      }
+      if (file != null) {
+        // TODO: Handle error, at least ignore in memory
+        try {
+          file.delete();
+        } catch (Exception e) {
+          logger.log(SentryLevel.ERROR, "Failed to delete.", e);
+        }
+      }
+    }
+  }
+
+  private void processEnvelope(SentryEnvelope envelope) throws IOException {
+    logger.log(SentryLevel.DEBUG, "Envelope for event Id: %s", envelope.getHeader().getEventId());
+    int items = 0;
+    for (SentryEnvelopeItem item : envelope.getItems()) {
+      items++;
+      if (item.getHeader() == null) {
+        logger.log(SentryLevel.ERROR, "Item %d has no header", items);
+        continue;
+      }
+      if ("event".equals(item.getHeader().getType())) {
+        Reader eventReader = null;
+        try {
+          eventReader = new InputStreamReader(new ByteArrayInputStream(item.getData()), UTF_8);
+          SentryEvent event = serializer.deserializeEvent(eventReader);
+          if (event == null) {
+            logger.log(
+                SentryLevel.ERROR,
+                "Item %d of type %s returned null by the parser.",
+                items,
+                item.getHeader().getType());
+          } else {
+            // TODO: Until sentry-native sends event_id in the header
+            //            if (envelope.getHeader().getEventId() != event.getEventId()) {
+            //              logger.log(
+            //                  SentryLevel.ERROR,
+            //                  "Item %d of has a different event id (%s) to the envelope header
+            // (s)",
+            //                  items,
+            //                  envelope.getHeader().getEventId(),
+            //                  event.getEventId());
+            //              continue;
+            //            }
+            hub.captureEvent(event);
```

Comment:
why is it the `hub` instead of `Sentry.captureEvent`?

Reply:
Testability is one reason. Sentry static is an entrypoint that will either clone the current hub on a new thread or use what's on TLS. I didn't see the need of doing that here, we can use the hub provided by the ctor.